src/generate-query.ts

```diff
@@ -610,6 +670,121 @@ function getArgsAndVars(
   }
 }
 
+function getListSizeDirective(field: FieldDefinitionNode): DirectiveNode {
+  return field?.directives.find((dir) => dir.name.value === 'listSize')
+}
+
+function getSlicingArguments(listSizeDirective: DirectiveNode): ArgumentNode {
+  if (!listSizeDirective) return undefined
+
+  const slicingArgumentsArg = listSizeDirective.arguments.find(
+    (arg) => arg.name.value === 'slicingArguments'
+  )
+
+  if (!slicingArgumentsArg || slicingArgumentsArg.value.kind !== 'ListValue')
+    return undefined
+
+  return slicingArgumentsArg
+}
+
+function getRequireOneSlicingArgument(
+  listSizeDirective: DirectiveNode
+): ArgumentNode {
+  if (!listSizeDirective) return undefined
+
+  const requireOneSlicingArg = listSizeDirective.arguments.find(
+    (arg) => arg.name.value === 'requireOneSlicingArgument'
+  )
+  if (
+    !requireOneSlicingArg ||
+    requireOneSlicingArg.value.kind !== 'BooleanValue'
+  )
+    return undefined
+
+  return requireOneSlicingArg
+}
+
+function getListSizeDirectiveDefaultValue(
+  listSizeDirective: DirectiveNode,
+  typeNode: TypeNode,
+  config: InternalConfiguration,
+  schema: GraphQLSchema
+) {
+  const slicingArgumentsArg = getSlicingArguments(listSizeDirective)
+  if (!slicingArgumentsArg) return undefined
+  let slicingArgumentsPaths = (slicingArgumentsArg.value as ListValueNode).values.map(
+    (value) => {
+      if (value.kind === 'StringValue') return value.value
+    }
+  )
+  const requireOneSlicingArg = getRequireOneSlicingArgument(listSizeDirective)
+  if (
+    !requireOneSlicingArg ||
+    (requireOneSlicingArg.value as BooleanValueNode).value === true
+  ) {
+    slicingArgumentsPaths = slicingArgumentsPaths.slice(0, 1)
+  }
+  const slicingArgumentsTokenizedPaths = slicingArgumentsPaths.map((value) =>
+    value.split('.')
+  )
+
+  return slicingArgumentsTokenizedPaths.reduce((res, tokenizedPath) => {
+    return merge.recursive(
```

Comment:
Doesn't this assume that all slicing arguments are input object types? That's not necessarily true right? It could just be a simple integer right?

Reply:
ahh yea, you are right. Good catch, I will fix this soon.

Reply:
fixed, resolving conversation